src/main/java/org/kuali/kfs/kew/actionrequest/service/ActionRequestService.java

```diff
@@ -0,0 +1,202 @@
+
+    void deactivateRequest(ActionTaken actionTaken, ActionRequest actionRequest,
+            ActivationContext activationContext);
+
+    void deactivateRequests(ActionTaken actionTaken, List actionRequests);
+
+    void deactivateRequests(ActionTaken actionTaken, List actionRequests, boolean simulate);
+
+    void deactivateRequests(ActionTaken actionTaken, List actionRequests, ActivationContext activationContext);
+
+    void deleteActionRequestGraph(ActionRequest actionRequest);
+
+    List findAllValidRequests(String principalId, String documentId, String requestCode);
+
+    List findAllValidRequests(String principalId, Collection actionRequests, String requestCode);
+
+    List<ActionRequest> findPendingByDoc(String documentId);
+
+    void saveActionRequest(ActionRequest actionRequest);
+
+    void activateRequest(ActionRequest actionRequest);
+
+    void activateRequest(ActionRequest actionRequest, boolean simulate);
+
+    void activateRequest(ActionRequest actionRequest, ActivationContext activationContext);
+
+    void activateRequests(Collection actionRequests);
+
+    void activateRequests(Collection actionRequests, boolean simulate);
+
+    void activateRequests(Collection actionRequests, ActivationContext activationContext);
+
+    List activateRequestNoNotification(ActionRequest actionRequest, boolean simulate);
+
+    List activateRequestNoNotification(ActionRequest actionRequest, ActivationContext activationContext);
+
+    ActionRequest findByActionRequestId(String actionRequestId);
+
+    List<ActionRequest> findPendingRootRequestsByDocId(String documentId);
+
+    List<ActionRequest> findPendingRootRequestsByDocIdAtRouteLevel(String documentId, Integer routeLevel);
+
+    List<ActionRequest> findPendingByDocIdAtOrBelowRouteLevel(String documentId, Integer routeLevel);
+
+    List<ActionRequest> findPendingRootRequestsByDocIdAtOrBelowRouteLevel(String documentId, Integer routeLevel);
+
+    List<ActionRequest> findPendingRootRequestsByDocumentType(String documentTypeId);
+
+    List<ActionRequest> findAllActionRequestsByDocumentId(String documentId);
+
+    List<ActionRequest> findAllRootActionRequestsByDocumentId(String documentId);
+
+    List<ActionRequest> findPendingByActionRequestedAndDocId(String actionRequestedCdCd, String documentId);
+
+    /**
+     * This method gets a list of ids of all principals who have a pending action request for a document.
+     */
+    List<String> getPrincipalIdsWithPendingActionRequestByActionRequestedAndDocId(String actionRequestedCd,
+            String documentId);
+
+    List<ActionRequest> findByStatusAndDocId(String statusCd, String documentId);
+
+    void alterActionRequested(List actionRequests, String actionRequestCd);
+
+    List<ActionRequest> findByDocumentIdIgnoreCurrentInd(String documentId);
+
+    List findActivatedByGroup(String groupId);
+
+    void updateActionRequestsForResponsibilityChange(Set<String> responsibilityIds);
```

Comment:
The KualiCo commit adds comments to this line.  Should they also be added?

/**
     * Schedule requeue of documents, affected by responsibility changes.
     *
     * @param responsibilityIds used to find documents to requeue.
     */

Reply:
comment added.